protoc-gen-openapiv2/internal/genopenapi/generator.go

```diff
@@ -143,21 +147,34 @@ func extensionMarshalJSON(so interface{}, extensions []extension) ([]byte, error
 }
 
 // encodeOpenAPI converts OpenAPI file obj to pluginpb.CodeGeneratorResponse_File
-func encodeOpenAPI(file *wrapper) (*descriptor.ResponseFile, error) {
-	var formatted bytes.Buffer
-	enc := json.NewEncoder(&formatted)
-	enc.SetIndent("", "  ")
-	if err := enc.Encode(*file.swagger); err != nil {
+func encodeOpenAPI(file *wrapper, format Format) (*descriptor.ResponseFile, error) {
+	// file.swagger depends on JSON pretty much, because of tags and
+	// extensions, so do the trick, always encode and decode JSON and
+	// only then use selected encoder.
```

Comment:
This does look a bit silly, and I wonder how much work it would be to add the tags where we have JSON tags today? What extensions are you talking about?

Reply:
I removed this part, added YAML tags to all struct fields. There is still `json.RawMessage` and structs with the `MarshalJSON` method.

I will mark the pull request as a draft until I do something with the rest.

Reply:
Fixed, it was not very easy, because sometimes structs used custom JSON decoding.